src/components/Clock.tsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+
+type State = {
+  date: Date,
+};
+
+type Props = {
+  name: number,
+};
+
+export class Clock extends React.Component<Props, State> {
+  state = {
+    date: new Date(),
+  };
+
+  timerId: NodeJS.Timeout = setInterval(() => {
+    this.setState({
+      date: new Date(),
+    });
+    // eslint-disable-next-line
+    console.log(this.state.date.toLocaleTimeString());
+  }, 1000);
+
+  componentDidMount() {
+    return this.timerId;
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    if (prevProps !== this.props) {
+      // eslint-disable-next-line no-console
+      console.log('The Clock was renamed from oldName to newName');
```

Comment:
Here should be numbers instead of `oldName` and `newName`
![image](https://user-images.githubusercontent.com/52370890/150788485-94ef5660-b111-4d38-b47c-750f7520fdac.png)